presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestMergeLimitWithDistinct.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.prestosql.sql.planner.iterative.rule.test.PlanBuilder;
+import io.prestosql.sql.planner.plan.DistinctLimitNode;
+import io.prestosql.sql.planner.plan.ValuesNode;
+import org.testng.annotations.Test;
+
+import java.util.function.Consumer;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.node;
+import static io.prestosql.sql.planner.iterative.rule.test.PlanBuilder.expression;
+
+public class TestMergeLimitWithDistinct
+        extends BaseRuleTest
+{
+    @Test
+    public void test()
+    {
+        tester().assertThat(new MergeLimitWithDistinct())
+                .on(p ->
+                        p.limit(
+                                1,
+                                p.aggregation(
+                                        (Consumer<PlanBuilder.AggregationBuilder>) aggregationBuilder -> {
```

Comment:
The type declaration for the lambda is unnecessary. You can also chain the invocations and shorten the variable name to reduce verbosity:

```java
p.aggregation(builder -> builder
        .singleGroupingSet(p.symbol("foo"))
        .source(p.values(p.symbol("foo"))))))
```